libsolidity/formal/CHC.cpp

```diff
@@ -262,7 +266,7 @@ void CHC::endVisit(FunctionDefinition const& _function)
 	{
 		auto sum = summary(_function);
 		auto ifacePre = smt::interfacePre(*m_interfaces.at(m_currentContract), *m_currentContract, m_context);
-		auto txConstraints = m_context.state().txConstraints(_function);
+		auto txConstraints = m_context.state().txConstraints(&_function);
```

Comment:
I guess this is because of API change in txConstraints that now requires a ptr instead of a reference? Could this be a different commit in this PR or is it non-separable from the aim stated in the PR title?

Reply:
Currently, it is needed to allow the use of function `txConstraints` also for the deployment phase where there is no explicit constructor.
But now I am thinking that we could split `txConstraints` to two methods, one that does require `FunctionDefinition` (adding constraints on function signature) and one that does not require it (only adding the type constrains on blockchain and transaction variables independent of the function being called). @leonardoalt, what do you think?

Reply:
Yea I think splitting could be a good idea. But the current version is also fine, I don't have a strong preference